### PR TITLE
feat: ex-Alert Groups are now creating incidents

### LINKED ISSUE #1501: [🔨 Enhancement]: Alert Groups should create incidents
As a user, when configuring Alert Groups rules, I want an **incident** created when the rule matches, **instead of the new alert** that is created as of today.

This work is mostly in the `RulesEngine` (rulesengine.py) at `run_rules` function. Happy to answer any question in our [Slack](https://slack.keephq.dev)

Extra requirements:

* Created incidents are not predictions, so `Incident.is_predicted` should be `False`
* Add flag "Create Incidents as Candidates" to rule config. If it's `True` set `Incident.is_confirmed` to `False`

Optional:
* Send an update to the UI using a [pusher](https://github.com/keephq/keep/blob/7060d721650511bfd39de7ecd010c414417713f6/keep/api/core/dependencies.py#L420) after creating new incidents

### PR DESCRIPTION
**Pull request by VladimirFilonov**
Closes #1501 

## 📑 Description
Alert Groups are no longer helpful after introducing Incident functionality but can still be useful as an incident automation tool. This PR adjusts Corellation rules to create Incidents instead of Alert Groups.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: keep-ui/components/navbar/NoiseReductionLinks.tsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ export const NoiseReductionLinks = ({ session }: NoiseReductionLinksProps) => {
       <Disclosure.Panel as="ul" className="space-y-2 p-2 pr-4">
         <li>
           <LinkWithIcon href="/rules" icon={Rules}>
-            <Subtitle>Alert Groups</Subtitle>
+            <Subtitle>Correlations</Subtitle>
```
Comment (Matvey-Kuk) [inline keep-ui/components/navbar/NoiseReductionLinks.tsx:43]: ❤️ !!!!!

File: keep/api/models/db/migrations/versions/2024-08-14-18-30_87594ea6d308.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+"""add rules-related fields to the incident
```
Comment (shahargl) [inline keep/api/models/db/migrations/versions/2024-08-14-18-30_87594ea6d308.py:1]: wouldn't it make sense to have one migration per PR?

**Comment by shahargl** [comment]
update docs - mainly 
- https://docs.keephq.dev/overview/ruleengine
- https://docs.keephq.dev/api-ref/groups/get-groups

File: keep-ui/app/rules/CorrelationTable.tsx [hunk 3]
```diff
@@ -94,14 +95,6 @@ export const CorrelationTable = ({ rules }: CorrelationTableProps) => {
       columnHelper.accessor("name", {
         header: "Correlation Name",
       }),
-      columnHelper.display({
```
Comment (shahargl) [inline keep-ui/app/rules/CorrelationTable.tsx:97]: why? how a user would have indication for if this rule works or not?

File: keep/api/models/db/migrations/versions/2024-08-14-18-30_87594ea6d308.py (hunk 2 above)
Comment (VladimirFilonov) [reply keep/api/models/db/migrations/versions/2024-08-14-18-30_87594ea6d308.py:1]: Going to squash them

File: keep-ui/app/rules/CorrelationTable.tsx (hunk 3 above)
Comment (VladimirFilonov) [reply keep-ui/app/rules/CorrelationTable.tsx:97]: These columns were always empty. I'm going to improve this UI in one of next steps

**Comment by VladimirFilonov** [comment]
> update docs - mainly
> 
> * https://docs.keephq.dev/overview/ruleengine
> * https://docs.keephq.dev/api-ref/groups/get-groups

Will do in the next iteration: https://github.com/keephq/keep/issues/1662

File: keep-ui/app/rules/CorrelationTable.tsx (hunk 3 above)
Comment (VladimirFilonov) [reply keep-ui/app/rules/CorrelationTable.tsx:97]: https://github.com/keephq/keep/issues/1664

**Comment by Matvey-Kuk** [comment]
A few notes from my testing:

- [ ] I think we should highlight that those "Correlations" are manual
Screenshot 2024-08-20 at 12 47 28

- [ ] In the incident page, now it's predicted by AI & Correlation Rules
Screenshot 2024-08-20 at 12 48 10

- [ ] If I set Correlation's "scan every" to "once per 5 seconds", it's slicing my alerts to batches of 5 seconds. I like the fact that it's appending to the same incident if I send alerts every 4 seconds, but I guess the config name is a bit misleading. Let's call it "Append to the same Incident if delay between alerts is below": 
Screenshot 2024-08-20 at 12 52 26

Screenshot 2024-08-20 at 12 49 01

- [ ] I was a bit confused by "34 alert(s) were found matching this condition", made me feel like correlation will work retrospectively, but it won't. Maybe nice to add some text that correlation will work for new alerts only, not for historical ones?

**Comment by VladimirFilonov** [comment]
> I was a bit confused by "34 alert(s) were found matching this condition", made me feel like correlation will work retrospectively, but it won't. Maybe worth adding some text that correlation will work for new alerts only, not for historical ones?

Maybe not the best UI solution, but visible =) Will think about better one, but out of this PR

image

**Review by Matvey-Kuk — APPROVED** [review]

**Review by Matvey-Kuk — APPROVED** [review]

**Pull request merged**